routing/cross_mwm_road_graph.hpp

```diff
@@ -88,14 +90,28 @@ inline string DebugPrint(BorderCross const & t)
 class CrossWeightedEdge
 {
 public:
-  CrossWeightedEdge(BorderCross const & target, double weight) : target(target), weight(weight) {}
+  CrossWeightedEdge(BorderCross const & target, double weight) : m_target(target), m_weight(weight)
+  {
+  }
+
+  inline BorderCross const & GetTarget() const { return m_target; }
+  inline double GetWeight() const { return m_weight; }
+
+  inline bool operator==(CrossWeightedEdge const & a) const
+  {
+    return m_target == a.m_target && m_weight == a.m_weight;
+  }
 
-  inline BorderCross const & GetTarget() const { return target; }
-  inline double GetWeight() const { return weight; }
+  inline bool operator<(CrossWeightedEdge const & a) const
```

Comment:
Could you please add a test that fails with old wrong comparer and passes with a correct one.